samples/bluetooth/encrypted_advertising/peripheral/src/peripheral_ead.c

```diff
@@ -0,0 +1,377 @@
+
+	err = bt_le_ext_adv_create(&params, &adv_cb, adv);
+	if (err) {
+		LOG_ERR("Failed to create advertiser (%d)", err);
+		return -1;
+	}
+
+	return 0;
+}
+
+static int start_adv(struct bt_le_ext_adv *adv)
+{
+	int err;
+	int32_t timeout = 0;
+	uint8_t num_events = 0;
+
+	struct bt_le_ext_adv_start_param start_params;
+
+	start_params.timeout = timeout;
+	start_params.num_events = num_events;
+
+	err = bt_le_ext_adv_start(adv, &start_params);
+	if (err) {
+		LOG_ERR("Failed to start advertiser (%d)", err);
+		return -1;
+	}
+
+	LOG_DBG("Advertiser started");
+
+	return 0;
+}
+
+static int stop_and_delete_adv(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	err = bt_le_ext_adv_stop(adv);
+	if (err) {
+		LOG_ERR("Failed to stop advertiser (err %d)", err);
+		return -1;
+	}
+
+	err = bt_le_ext_adv_delete(adv);
+	if (err) {
+		LOG_ERR("Failed to delete advertiser (err %d)", err);
+		return -1;
+	}
+
+	return 0;
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err) {
+		LOG_ERR("Failed to connect to %s (%u)", addr, err);
+		return;
+	}
+
+	LOG_DBG("Connected to %s", addr);
+
+	default_conn = conn;
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("Disconnected from %s (reason 0x%02x)", addr, reason);
+
+	k_poll_signal_raise(&disconn_signal, 0);
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (!err) {
+		LOG_DBG("Security changed: %s level %u", addr, level);
+	} else {
+		LOG_DBG("Security failed: %s level %u err %d", addr, level, err);
+	}
+}
+
+static void auth_passkey_confirm(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Confirm passkey for %s: %s", addr, passkey_str);
+
+	k_poll_signal_raise(&passkey_enter_signal, 0);
+}
+
+static void auth_passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	char passkey_str[7];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	snprintk(passkey_str, 7, "%06u", passkey);
+
+	LOG_DBG("Passkey for %s: %s", addr, passkey_str);
+}
+
+static void auth_cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_DBG("Pairing cancelled: %s", addr);
+}
+
+static int update_security(void)
+{
+	int err;
+
+	await_signal(&passkey_enter_signal);
+
+	err = bt_conn_auth_passkey_confirm(default_conn);
+	if (err) {
+		LOG_DBG("Failed to confirm passkey.");
+		return -1;
+	}
+
+	return 0;
+}
```

Comment:
If you are just autonomously confirm the passkey then you are not confirming the passkey at all. This sets a very poor precedence for a sample application, and should definitely not be merged. 

If you want to confirm the passkey, a user needs to actively do that.

Reply:
I think you're right. But this make testing the sample harder. What do you think if the test use its own `update_security` function and do it automatically but when running the sample it require user interaction?

Reply:
>  But this make testing the sample harder.

As mentioned earlier, this is a _sample_ and not a _test_. 

You may be able to define the sample in a way that user input is required to do the passkey confirmation (e.g. press a button or something like that), and then have a way to also confirm the passkey from the actual babblesim test

Reply:
Yes I agree, I suggest to use different `update_security` function if it is from BabbleSim or not. Do you think it's the right approach or do you suggest something else?